### PR TITLE
ci: Add support for ARM/RISCV by QEMU

### PR DESCRIPTION
**Pull request by no1wudi**

### CODE REVIEW

File: tests/wamr-test-suites/spec-test-script/runtest.py [hunk 1]
```diff
@@ -824,6 +838,8 @@ def test_assert_trap(r, opts, form):
 
     elif not m and n:
         module = n.group(1)
+        module = "/tmp/"+module
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply tests/wamr-test-suites/spec-test-script/runtest.py:841]: OK, I'll try to get tmp dir by standard python API

File: product-mini/platforms/posix/main.c [hunk 2]
```diff
@@ -158,7 +161,7 @@ app_instance_repl(wasm_module_inst_t module_inst)
                 cmd[n - 1] = '\0';
         }
         if (!strcmp(cmd, "__exit__")) {
-            printf("exit repl mode\n");
+            puts("exit repl mode\n");
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply product-mini/platforms/posix/main.c:164]: May be we can revert this change.

File: tests/wamr-test-suites/spec-test-script/all.py [hunk 3]
```diff
@@ -105,7 +103,10 @@ def test_case(
     xip_flag=False,
     clean_up_flag=True,
     verbose_flag=True,
+    qemu_flag=False,
+    qemu_firmware='',
 ):
+    global IWASM_CMD
```
Comment (lum1n0us) [inline tests/wamr-test-suites/spec-test-script/all.py:109]: Don't suggest to change `IWASM_CMD`.

Maybe adding a new `QEUMU_CMD` instead ? and change L128 like 
`CMD.append(QEMU_CMD if qemu_flag else (IWASM_CMD if not sgx_flag else IWASM_SGX_CMD))`

Comment (no1wudi) [reply tests/wamr-test-suites/spec-test-script/all.py:109]: OK

**Comment by wenyongh** [comment]
Run failed in MacOS:
./test_wamr.sh -s spec
```
Starting interpreter for module '/var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpk0m_eu7m.wasm'
Running: ../../../product-mini/platforms/darwin/build/iwasm --heap-size=0 -v=0 --repl /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpk0m_eu7m.wasm
Testing(return) call-overwritten-element  = 0x42:i32
;; Element sections across multiple modules change the same table
Writing WAST module to '/var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmp7sscdv2a.wast'
Compiling WASM to '/var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpqekvp4rl.wasm'
Running: ./wabt/out/gcc/Release/wat2wasm --enable-thread --no-check /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmp7sscdv2a.wast -o /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpqekvp4rl.waTraceback (most recent call last):
  File "/Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/./all.py", line 464, in <module>
sm
Starting interpreter for module '/var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpqekvp4rl.wasm'
Running: ../../../product-mini/platforms/darwin/build/iwasm --heap-size=0 -v=0 --repl /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/tmpqekvp4rl.wasm
Testing Trap Mode: /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/module1
Starting interpreter for module '/var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/module1.wasm'
Running: ../../../product-mini/platforms/darwin/build/iwasm --heap-size=0 -v=0 --repl /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/module1.wasm
Did not one of following prompt(s): ['webassembly> ']
    Got      : ''
THE FINAL EXCEPTION IS Failed:
  expected: 'SystemExit(1)'
  got: 'Read file to buffer failed: open file /var/folders/sd/mcflkwfs05l2myvkw51j1q480000gp/T/module1.wasm failed.
'
Removing tempfiles
### End testing /Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/spec/test/core/elem.wast
failed with a non-zero return code 101
    sys.exit(0 if main() else 1)
  File "/Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/./all.py", line 420, in main
    ret = test_suite(
  File "/Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/./all.py", line 301, in test_suite
    raise e
  File "/Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/./all.py", line 284, in test_suite
    test_case(
  File "/Users/wamr/WASM/wamr-internal-test/wamr-github/tests/wamr-test-suites/workspace/./all.py", line 202, in test_case
    raise Exception(case_name)
Exception: elem
```

**Comment by no1wudi** [comment]
@wenyongh Could you try it again or we should add macos to CI

**Comment by wenyongh** [comment]
> @wenyongh Could you try it again or we should add macos to CI

Yes, it works now. Had better not add MacOS spec test to CI, it takes long time to runtime spec case since `-P ` option isn't supported.

**Review by wenyongh — APPROVED** [review]
LGTM

**Pull request merged**